File: backend-python/src/controllers/health_controller.py
from flask import jsonify


class HealthController:
    def __init__(self, repository):
        self.repository = repository

    def check(self):
        try:
            self.repository.get_all()
            return jsonify({"status": "ok"}), 200
        except Exception:
            return jsonify({"status": "unavailable"}), 503

File: backend/src/controllers/healthController.js
function createHealthController(repository) {
  async function check(req, res) {
    try {
      await repository.getAll();
      return res.status(200).json({ status: 'ok' });
    } catch {
      return res.status(503).json({ status: 'unavailable' });
    }
  }

  return { check };
}

module.exports = {
  createHealthController,
};
